tensorflow/lite/kernels/conv.cc

```diff
@@ -168,8 +216,8 @@ static TfLiteStatus AllocateTemporaryTensorsIfRequired(TfLiteContext* context,
   // buffer to store the results.
   // This path is only used for float processing, so only create the buffer if
   // we're running with that data type.
-  data->need_hwcn_weights = (input->type == kTfLiteFloat32 &&
-                             data->run_multithreaded_kernel && !is_hybrid);
```

Comment:
Some changes were recently made to this class to avoid im2col generation, can you rebase?

Reply:
@jdduke : I have gone through latest merged changes. I think with the new changes kReference block in EvalFloat() method becomes dead code. As it always hits the else case in Eval() which hardcode kernel to kGenericOptimized. Please confirm whether this behavior is intended, Thanks!

I have already rebased with new changes, but waiting for your confirmation on this point, post your confirmation will upstream rebased changes.